src/core/Cline.ts

```diff
@@ -2568,13 +2568,20 @@ export class Cline {
 									?.mcpHub?.callTool(server_name, tool_name, parsedArguments)
 
 								// TODO: add progress indicator and ability to parse images and non-text responses
+								const supportsImages = this.api.getModel().info.supportsImages ?? false
 								const toolResultPretty =
 									(toolResult?.isError ? "Error:\n" : "") +
 										toolResult?.content
 											.map((item) => {
 												if (item.type === "text") {
 													return item.text
 												}
+												if (item.type === "image" && supportsImages) {
+													return ` data:${item.mimeType};base64,${item.data} `
+												}
+												if (item.type === "image" && !supportsImages) {
+													return "[Image placeholder (images not supported)]"
+												}
```

Comment:
We can consolidate the conditions here a bit.
```suggestion
																								if (item.type === "image") {
													if (supportsImages) {
														return ` data:${item.mimeType};base64,${item.data} `
													}
													return "[Image placeholder (images not supported)]"
												}
```